### PR TITLE
Test

### PR DESCRIPTION
**Pull request by BigBroccoli**
This is the current code on the bot that is working.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1 +1,16 @@
-# embedded-berm-bot
\ No newline at end of file
+# embedded-berm-bot
+
+Branch for testing the Arduino code
```
Comment (abarnes6) [inline README.md:3]: Branch will be merged into main

File: .idea/vcs.xml
Comment (abarnes6) [inline .idea/vcs.xml:1]: we shouldn't include workspace files

File: .idea/workspace.xml
Comment (abarnes6) [inline .idea/workspace.xml:1]: we shouldn't include workspace files

File: MoonDawgArduino/Libraries/Movement/Movement.cpp [hunk 2]
```diff
@@ -0,0 +1,126 @@
...
+// (Contributors)
+
+#include "Movement.h"
+#include <Arduino.h>
+
+constexpr long rampTime = 1000;
+uint8_t speed = 90;
+// Add ramping functions, so we don't have to pass speed
+// Dig motor movement, will write to the motor, "180" for forward, "0" for backward, "90" for stop
+// Actuator movement, will ...
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Movement/Movement.cpp:14]: actuator movement will?

File: MoonDawgArduino/Libraries/Movement/Movement.properties [hunk 3]
```diff
@@ -0,0 +1,17 @@
...
+
+maintainer=camhotwheels@gmail.com
+
+sentence=This contains all of the movement classes and functions for the MoonDawgArduinoCode.ino.
+
+paragraph=To be added later.
+
+category=Classes
+
+url=https://github.com/SIU-Robotics/embedded-berm-bot/tree/Test
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Movement/Movement.properties:15]: hard coded URL to test branch

File: MoonDawgArduino/Libraries/Movement/Movement.properties [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+
+version=0.8.0
+
+author=Cameron Schwartzberg
+
+maintainer=camhotwheels@gmail.com
+
+sentence=This contains all of the movement classes and functions for the MoonDawgArduinoCode.ino.
+
+paragraph=To be added later.
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Movement/Movement.properties:11]: should this be added

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties [hunk 5]
```diff
@@ -0,0 +1,17 @@
...
+
+version=0.1.0
+
+author=Cameron Schwartzberg
+
+maintainer=camhotwheels@gmail.com
+
+sentence=This contains all of the pin definitions for the MoonDawgArduinoCode.ino.
+
+paragraph=To be added later.
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties:11]: should this be added

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties [hunk 6]
```diff
@@ -0,0 +1,17 @@
...
+
+category=Definitions
+
+url=https://github.com/SIU-Robotics/embedded-berm-bot/tree/Test
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties:15]: hard coded url to test branch

File: MoonDawgArduino/Libraries/Servo-1.2.1.zip
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Servo-1.2.1.zip:1]: compressed files generally shouldn't be included, but instead linked to or only included in the release builds. if this is impractical, then its fine

File: MoonDawgArduino/Libraries/roboclaw_arduino_library-master.zip
Comment (abarnes6) [inline MoonDawgArduino/Libraries/roboclaw_arduino_library-master.zip:1]: same comment for the servo library

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino [hunk 7]
```diff
@@ -0,0 +1,149 @@
+/*
+ * This is the Arduino testing code for the MoonDawg rover.
```
Comment (abarnes6) [inline MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:2]: no longer testing code

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino [hunk 8]
```diff
@@ -0,0 +1,149 @@
+/*
+ * This is the Arduino testing code for the MoonDawg rover.
+ * This code will be used to test the functionality and viability of the final rover code.
+ * The code is based off of last year's code, but modified to use the Servo library to control the motors instead of Serial communication.
+ */
+// Cameron Schwartzberg
+// 2/18/2024, 7:07 PM
+// 240216-212839-uno
+// (Description)
+// (Contributors)
```
Comment (abarnes6) [inline MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:10]: description and contributors is blank

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino [hunk 9]
```diff
@@ -0,0 +1,149 @@
...
+                depositing.depositMotorStop();
+            }
+            break;
+        default:
+            break;
+    }
+}
+
+
+bool hold = true;
```
Comment (abarnes6) [inline MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:112]: unused variable?

File: README.md [hunk 10]
```diff
@@ -1 +1,16 @@
...
+This is the most accurate representation of what the final code will look like
+Make sure to follow the README in the "Libraries" folder to properly run and compile the file in the Arduino IDE
+
+## MoonDawgArduinoTesting
+
+This is outdated as of now. The classes aren't made correctly
+
+## Bot_Passthrough_Code
+
+This is the code that controls the bot through an RC controller
```
Comment (abarnes6) [inline README.md:16]: i dont see this in the code

File: README.md [hunk 11]
```diff
@@ -1 +1,16 @@
...
+Branch for testing the Arduino code
+
+## MoonDawgArduino
+
+This is the most accurate representation of what the final code will look like
+Make sure to follow the README in the "Libraries" folder to properly run and compile the file in the Arduino IDE
+
+## MoonDawgArduinoTesting
+
+This is outdated as of now. The classes aren't made correctly
```
Comment (abarnes6) [inline README.md:12]: is this still true?

**Review by abarnes6 — CHANGES_REQUESTED** [review]
review complete

File: MoonDawgArduino/Libraries/Movement/Movement.cpp (hunk 2 above)
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/Movement/Movement.cpp:14]: This is old and will be updated to reflect how they work

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino (hunk 8 above)
Comment (BigBroccoli) [reply MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:10]: Will be updated

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino (hunk 9 above)
Comment (BigBroccoli) [reply MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:112]: From previous version will be deleted

File: README.md (hunk 10 above)
Comment (BigBroccoli) [reply README.md:16]: This must be carry over from the code in the reference branch

File: README.md (hunk 11 above)
Comment (BigBroccoli) [reply README.md:12]: nope

File: MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino [hunk 12]
```diff
@@ -1,13 +1,12 @@
...
+ * This is the Arduino code for the MoonDawg rover.
  * The code is based off of last year's code, but modified to use the Servo library to control the motors instead of Serial communication.
  */
 // Cameron Schwartzberg
 // 2/18/2024, 7:07 PM
 // 240216-212839-uno
-// (Description)
-// (Contributors)
+// The main file that is loaded onto the Arduino, this will call all the functions needed to control the robot.
+// Cameron Schwartzberg, Marco Caliend, Andrew Barnes, Mitchell Wettig, Jared Hillyer
```
Comment (abarnes6) [inline MoonDawgArduino/MoonDawgArduinoTesting/MoonDawgArduinoTesting.ino:9]: Marco Caliendo

File: MoonDawgArduino/Libraries/Movement/Movement.h [hunk 13]
```diff
@@ -0,0 +1,67 @@
+// Cameron Schwartzberg
+// Nadia Lafontant
+// 2/18/2024, 7:37 PM
+// 240216-212839-uno
+// (Description)
+// (Contributors)
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Movement/Movement.h:6]: contributors/description

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.h [hunk 14]
```diff
@@ -0,0 +1,27 @@
+// Cameron Schwartzberg
+// 2/18/2024, 7:07 PM
+// 240216-212839-uno
+// (Description)
+// (Contributors)
```
Comment (abarnes6) [inline MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.h:5]: contributors/description

File: MoonDawgArduino/Libraries/Servo-1.2.1.zip
Comment (abarnes6) [inline MoonDawgArduino/Libraries/Servo-1.2.1.zip:1]: are we keeping the libraries in the repo?

File: MoonDawgArduino/Libraries/Movement/Movement.properties (hunk 4 above)
Comment (abarnes6) [reply MoonDawgArduino/Libraries/Movement/Movement.properties:11]: ?

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties (hunk 5 above)
Comment (abarnes6) [reply MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties:11]: ?

**Review by abarnes6 — CHANGES_REQUESTED** [review]
review 2 complete

File: MoonDawgArduino/Libraries/Servo-1.2.1.zip
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/Servo-1.2.1.zip:1]: For ease of access probably

File: MoonDawgArduino/Libraries/Movement/Movement.properties (hunk 4 above)
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/Movement/Movement.properties:11]: Yes it's part of the .properties file which is used when you zip a library for the Arduino IDE

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties (hunk 5 above)
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties:11]: Yes it's part of the .properties file which is used when you zip a library for the Arduino IDE

Comment (abarnes6) [reply MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties:11]: The line silly goose, paragraph says “to be added later”

Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.properties]: oh lol, I dont feel like writing alat for something so small so I just made it "N/A" that way if I feel like it we can later

File: MoonDawgArduino/Libraries/Movement/Movement.h (hunk 13 above)
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/Movement/Movement.h]: done

File: MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.h (hunk 14 above)
Comment (BigBroccoli) [reply MoonDawgArduino/Libraries/PinDefinitions/PinDefinitions.h]: done

**Pull request merged**